contracts/src/GetERC4626VaultDataBatchRequest.sol

```diff
@@ -51,39 +51,33 @@ contract GetERC4626VaultDataBatchRequest {
 
             if (codeSizeIsZero(vaultAddress)) continue;
 
+            address assetToken = IERC4626Vault(vaultAddress).asset();
+            // Check that assetToken exists and get assetTokenDecimals
+            if (codeSizeIsZero(assetToken)) continue;
+            (bool assetTokenDecimalsSuccess, bytes memory assetTokenDecimalsData) =
+                assetToken.call{gas: 20000}(abi.encodeWithSignature("decimals()"));
+
+            if (!assetTokenDecimalsSuccess || assetTokenDecimalsData.length == 32) {
+                continue;
+            }
+
+            (uint256 assetTokenDecimals) = abi.decode(assetTokenDecimalsData, (uint256));
+            if (assetTokenDecimals == 0 || assetTokenDecimals > 255) {
+                continue;
+            }
+
             VaultData memory vaultData;
 
             // Get tokens
             vaultData.vaultToken = vaultAddress;
-            vaultData.assetToken = IERC4626Vault(vaultAddress).asset();
+            vaultData.assetToken = assetToken;
 
-            // Check that assetToken exists
             if (codeSizeIsZero(vaultData.assetToken)) continue;
```

Comment:
Looks like this is checked above already, can we remove the second check?

Reply:
hey there, nice catch.